core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -938,6 +941,97 @@ public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder) {
         throw new IllegalArgumentException("weighting " + weighting + " not supported");
     }
 
+    private GHIntHashSet getBlockedEdges(HintsMap hints, FlagEncoder encoder) {
```

Comment:
I had exactly the same thought :), this is what I meant with:

> Add a (helper)class that searches the Graph for edges that are in a BBox, or maybe circle, or ...

Will set up something :)